### DISCUSSION TITLE
how to write mdx in mkdocs material?

### DISCUSSION THREAD

**Discussion by Thecode764** [squidfunk/mkdocs-material#7309] [Q&A] [answered]
hi. how to write mdx in mkdocs material?

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @Thecode764,
if you mean this https://mdxjs.com/ then this is a JavaScript project, not available for Python.

[Python-Markdown](https://github.com/Python-Markdown/markdown) has support for extensions that add similar syntax functionalities, but still it's not the same, as MkDocs uses Jinja2 for templating, everything apart of the content.
See for example:
- https://facelessuser.github.io/pymdown-extensions/
- https://facelessuser.github.io/pymdown-extensions/extensions/emoji/
- https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/

If you really need to use MDX then better change the docs generator to a JavaScript-based, like Docosaurus:
- https://squidfunk.github.io/mkdocs-material/alternatives/

**Reply by kamilkrzyskow** [reply]
Actually this question was asked before @Thecode764, please search the discussion boards before asking questions.
Closing and locking as duplicate of:
- https://github.com/squidfunk/mkdocs-material/discussions/3621
